src/app/clusters/on-off-server/on-off-server.h

```diff
@@ -92,23 +92,29 @@ class OnOffServer
 struct OnOffEffect
 {
     using OffWithEffectTriggerCommand = void (*)(OnOffEffect *);
+    static_assert(std::is_same<std::underlying_type_t<chip::app::Clusters::OnOff::OnOffDelayedAllOffEffectVariant>,
+                               std::underlying_type_t<chip::app::Clusters::OnOff::OnOffDyingLightEffectVariant>>::value,
+                  "chip::app::Clusters::OnOff::OnOffDelayedAllOffEffectVariant and "
+                  "chip::app::Clusters::OnOff::OnOffDyingLightEffectVariant underlying types differs.");
 
     chip::EndpointId mEndpoint;
     OffWithEffectTriggerCommand mOffWithEffectTrigger = nullptr;
     chip::app::Clusters::OnOff::OnOffEffectIdentifier mEffectIdentifier;
-    uint8_t mEffectVariant;
+    std::underlying_type_t<chip::app::Clusters::OnOff::OnOffDelayedAllOffEffectVariant> mEffectVariant;
     OnOffEffect * nextEffect = nullptr;
 
-    OnOffEffect(
-        chip::EndpointId endpoint, OffWithEffectTriggerCommand offWithEffectTrigger,
-        chip::app::Clusters::OnOff::OnOffEffectIdentifier effectIdentifier = EMBER_ZCL_ON_OFF_EFFECT_IDENTIFIER_DELAYED_ALL_OFF,
-
-        /*
-         * effectVariant's type depends on the effect effectIdentifier so we don't know the type at compile time.
-         * Casting to uint8_t for more flexibility since the type can be OnOffDelayedAllOffEffectVariant or
-         * OnOffDelayedAllOffEffectVariant
-         */
-        uint8_t effectVariant = static_cast<uint8_t>(EMBER_ZCL_ON_OFF_DELAYED_ALL_OFF_EFFECT_VARIANT_FADE_TO_OFF_IN_0P8_SECONDS));
+    OnOffEffect(chip::EndpointId endpoint, OffWithEffectTriggerCommand offWithEffectTrigger,
+                chip::app::Clusters::OnOff::OnOffEffectIdentifier effectIdentifier =
+                    chip::app::Clusters::OnOff::OnOffEffectIdentifier::kDelayedAllOff,
+
+                /*
+                 * effectVariant's type depends on the effect effectIdentifier so we don't know the type at compile time.
+                 * The assertion at the beginning of this method ensure the effect variants shared the same base type.
+                 * Casting to the common base type for more flexibility since the type can be OnOffDelayedAllOffEffectVariant or
+                 * OnOffDelayedAllOffEffectVariant
+                 */
+                std::underlying_type_t<chip::app::Clusters::OnOff::OnOffDelayedAllOffEffectVariant> effectVariant =
```

Comment:
How about we `using EffectVariantType = std::underlying_type_t<chip::app::Clusters::OnOff::OnOffDelayedAllOffEffectVariant>` or something and use that both here and for the member?  Might be more readable.

Reply:
I end up using that. This is much more readable. Thanks!